src/libxrpl/protocol/STTx.cpp

```diff
@@ -308,80 +336,70 @@ STTx::getMetaSQL(
         getFieldU32(sfSequence) % inLedger % status % rTxn % escapedMetaData);
 }
 
-Expected<void, std::string>
-STTx::checkSingleSign(RequireFullyCanonicalSig requireCanonicalSig) const
+static Expected<void, std::string>
+singleSignHelper(
+    STObject const& signer,
+    Slice const& data,
+    STTx::RequireFullyCanonicalSig requireCanonicalSig,
```

Comment:
I think this param can be removed, since `rules` is a global variable now.

That might get complicated, though, so I'm fine if you don't want to touch that in this PR.